src/arithmetic/boolean_funcs/boolean_funcs.c

```diff
@@ -79,11 +79,9 @@ SIValue AR_GT(SIValue *argv, int argc, void *private_data) {
 	if(disjointOrNull == COMPARED_NULL) {
 		// Comparisons with NULL values always return NULL.
 		return SI_NullVal();
-	} else if(disjointOrNull == DISJOINT) {
-		// Emit error when attempting to compare invalid types
-		Error_SITypeMismatch(b, SI_TYPE(a));
-		return SI_NullVal(); // The return doesn't matter, as the caller will check for errors.
 	}
+	if(disjointOrNull == COMPARED_NAN) return SI_BoolVal(false);
+	if(disjointOrNull == DISJOINT) return SI_NullVal();
```

Comment:
this line can be merged with: `if(disjointOrNull == COMPARED_NULL) {`
please add a comment explaining the situation similar to line 80